retrieval/scrape.go

```diff
@@ -434,7 +443,8 @@ func newScrapeLoop(ctx context.Context, sc scraper, app, reportApp func() storag
 		scraper:        sc,
 		appender:       app,
 		reportAppender: reportApp,
-		cache:          map[string]uint64{},
+		refCache:       map[string]uint64{},
+		lsetCache:      map[uint64]lsetCacheEntry{},
```

Comment:
It looks like these caches are slow memory leaks if long-lived targets change series a lot. Maybe at least add a TODO comment to periodically purge them or something? Actually maybe the same staleness logic can be reused to purge stale cache entries?